Sources/MongoSwift/BSON/BsonValue.swift

```diff
@@ -365,22 +327,26 @@ public struct Decimal128: BsonValue, Equatable, Codable {
 
 /// An extension of `Double` to represent the BSON Double type.
 extension Double: BsonValue {
+
     public var bsonType: BsonType { return .double }
+
     public func encode(to storage: DocumentStorage, forKey key: String) throws {
         if !bson_append_double(storage.pointer, key, Int32(key.count), self) {
             throw bsonEncodeError(value: self, forKey: key)
         }
     }
 
-    public static func from(iter: inout bson_iter_t) -> BsonValue {
-        return bson_iter_double(&iter)
+    public init(from iter: DocumentIterator) throws {
+        self = bson_iter_double(&iter.iter)
```

Comment:
Why isn't `self.init()` used here, as with other types? Likewise a question for Int and other primitive types below.